### PR TITLE
UGENE-7161 Assert in "Find Groups of annotated regions" dialog

### PR DESCRIPTION
**Pull request by rasputinkirill**

### CODE REVIEW

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: nit: use {} instead of QStringList()

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: nit: do not repeat 'virtual' with 'override'

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Remove this release. Independent press/release events must be used with extra care because they may have cross test issues

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: remove 'virtual'

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: plusButton is never null

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: use GTWidget: getActveModalWidget(os) that is never null

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: nit: use auto - do not repeat the type twice

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: toolbar is never null

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: use auto + farButton is never null

File: src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp
Comment (yalgaer) [inline src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp]: Give the class a meaningful name

File: src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp
Comment (yalgaer) [inline src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp]: this won't build

Comment (yalgaer) [reply src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp]: Do you really want to return here and skip all other 'items'?

File: src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp
Comment (yalgaer) [inline src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp]: Same here: what about other items?

File: src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp
Comment (yalgaer) [inline src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp]: Do you really want to return here?

**Review by yalgaer — CHANGES_REQUESTED** [review]
Please check the comments

**Review by smlnkcv — COMMENTED** [review]
Works well, no assert

**Comment by rasputinkirill** [comment]
About CHECKs, I've just replaced all asserts with them. Because after introduce SAFW_POINT and CHECK we try to get rid with asserts. This file is small, there is no tasks, there is only 'silent' algorithm and there are not so many asserts, so i decided to replace them with CHECKS. 
Do you think it is better to return asserts back?

**Comment by yalgaer** [comment]
>  Do you think it is better to return asserts back?

Assertions (aka SAFE_POINTS) are not the same as CHECKS
CHECK is an expected situation. It is a 1-line helper that replaces 3 boilerplate if () {} lines
SAFE_POINT (assert) is a bug in the UGENE

The good solution is to replace asserts with SAFE_POINTS and fix the related bugs.
I think you are aware that with CHECKs you only hide the bugs and interrupt the computation (items in the rest of the list are not processed at all)

File: src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp
Comment (yalgaer) [inline src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp]: nit: the change the comment to the opposite: item region is not in the search region.

File: src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp
Comment (yalgaer) [inline src/plugins/annotator/src/CollocationsSearchAlgorithm.cpp]: same here: change message to the opposite

**Review by yalgaer — COMMENTED** [review]
This fix looks much better than before. Lets wait until tests before approval.

**Review by yalgaer — APPROVED** [review]

**Pull request merged**
